### PR TITLE
machine_management: add note about using original boot media

### PR DESCRIPTION
**Pull request by miabbott**
We don't support using newer boot media to add nodes to a clusters.

### CODE REVIEW

**Comment by miabbott** [comment]
cc: @ashcrow @bobfuru @yuqi-zhang @kikisdeliveryservice

**Comment by miabbott** [comment]
This should be added to the 4.6 docs, too

**Review by yuqi-zhang — APPROVED** [review]
Note that this isn't just 4.6, we've technically never "supported" updating boot media directly in any way. It just so happens that the 4.6 ignition spec upgrade is the first time we've updated the pointer config

**Comment by kikisdeliveryservice** [comment]
> Note that this isn't just 4.6, we've technically never "supported" updating boot media directly in any way. It just so happens that the 4.6 ignition spec upgrade is the first time we've updated the pointer config

Maybe worth being extra clear then and just saying to use the original boot media wiht no 4.6 caveat

**Review by ashcrow — APPROVED** [review]

**Comment by miabbott** [comment]
> Maybe worth being extra clear then and just saying to use the original boot media wiht no 4.6 caveat

Pushed an update that was more clear

**Review by ashcrow — APPROVED** [review]

File: machine_management/user_infra/adding-bare-metal-compute-user-infra.adoc [hunk 1]
```diff
@@ -15,6 +15,18 @@ create your cluster. If you do not have these files, you must obtain them by
 following the instructions in the
 xref:../../installing/installing_bare_metal/installing-bare-metal.adoc#installing-bare-metal[installation procedure].
 
+[IMPORTANT]
+===
+When adding additional compute machines to your {product-title} cluster, you should use the
+{op-system} boot media that matches the same minor version that was used to originally
+install the {product-title} cluster.  For example, if you originally installed
+{product-title} 4.4, you should add additional compute machines using {op-system} 4.4
```
Comment (kalexand-rh) [inline machine_management/user_infra/adding-bare-metal-compute-user-infra.adoc:23]:
```suggestion
{product-title} 4.4, you must add additional compute machines using {op-system} 4.4
```

File: machine_management/user_infra/adding-bare-metal-compute-user-infra.adoc [hunk 2]
```diff
@@ -15,6 +15,18 @@ create your cluster. If you do not have these files, you must obtain them by
...
+boot media.
+
+It is not supported to add additional compute machines to your cluster using newer
+{op-system} boot media.
+===
```
Comment (kalexand-rh) [inline machine_management/user_infra/adding-bare-metal-compute-user-infra.adoc:28]: Please add a sentence about upgrading the machines after you add them.

**Review by kalexand-rh — COMMENTED** [review]
I have a few requests, but this is looking good.

File: machine_management/user_infra/adding-bare-metal-compute-user-infra.adoc [hunk 3]
```diff
@@ -15,6 +15,18 @@ create your cluster. If you do not have these files, you must obtain them by
...
 
+[IMPORTANT]
+===
+When adding additional compute machines to your {product-title} cluster, you should use the
+{op-system} boot media that matches the same minor version that was used to originally
+install the {product-title} cluster.  For example, if you originally installed
+{product-title} 4.4, you should add additional compute machines using {op-system} 4.4
+boot media.
+
+It is not supported to add additional compute machines to your cluster using newer
```
Comment (bobfuru) [inline machine_management/user_infra/adding-bare-metal-compute-user-infra.adoc:26]: Adding additional compute machines to your cluster using newer {op-system} boot media is not supported.
```suggestion
It is not supported to add additional compute machines to your cluster using newer
```

File: machine_management/user_infra/adding-vsphere-compute-user-infra.adoc [hunk 4]
```diff
@@ -12,6 +12,18 @@ vSphere.
 
 * You xref:../../installing/installing_vsphere/installing-vsphere.adoc#installing-vsphere[installed a cluster on vSphere].
 
+[IMPORTANT]
```
Comment (bobfuru) [inline machine_management/user_infra/adding-vsphere-compute-user-infra.adoc:15]: Same comments apply for this module as well.

**Review by bobfuru — CHANGES_REQUESTED** [review]
Added a suggestion to Kathryn's. Thanks!

**Review by bobfuru — APPROVED** [review]

**Comment by miabbott** [comment]
Thanks @kalexand-rh @bobfuru; pushed update with your feedback

File: machine_management/user_infra/adding-bare-metal-compute-user-infra.adoc [hunk 5]
```diff
@@ -15,6 +15,19 @@ create your cluster. If you do not have these files, you must obtain them by
 following the instructions in the
 xref:../../installing/installing_bare_metal/installing-bare-metal.adoc#installing-bare-metal[installation procedure].
 
+[IMPORTANT]
+===
+When adding additional compute machines to your {product-title} cluster, use the
+{op-system} boot media that matches the same minor version that was used to
+install the {product-title} cluster.  For example, if you installed
```
Comment (bobfuru) [inline machine_management/user_infra/adding-bare-metal-compute-user-infra.adoc:22]: cluster. For example,
```suggestion
install the {product-title} cluster. For example, if you installed
```

File: machine_management/user_infra/adding-bare-metal-compute-user-infra.adoc [hunk 6]
```diff
@@ -15,6 +15,19 @@ create your cluster. If you do not have these files, you must obtain them by
...
+{product-title} 4.4, you must add additional compute machines using {op-system} 4.4
+boot media.
+
+Adding additional compute machines to your cluster using newer {op-system} boot
+media is not supported.  After adding the compute machines, they will automatically
```
Comment (bobfuru) [inline machine_management/user_infra/adding-bare-metal-compute-user-infra.adoc:27]: supported.  After
```suggestion
media is not supported. After adding the compute machines, they will automatically
```

File: machine_management/user_infra/adding-vsphere-compute-user-infra.adoc [hunk 7]
```diff
@@ -12,6 +12,19 @@ vSphere.
...
+===
+When adding additional compute machines to your {product-title} cluster, use the
+{op-system} boot media that matches the same minor version that was used to
+install the {product-title} cluster.  For example, if you installed
```
Comment (bobfuru) [inline machine_management/user_infra/adding-vsphere-compute-user-infra.adoc:19]: cluster. After
```suggestion
install the {product-title} cluster. For example, if you installed
```

File: machine_management/user_infra/adding-vsphere-compute-user-infra.adoc [hunk 8]
```diff
@@ -12,6 +12,19 @@ vSphere.
...
+{product-title} 4.4, you must add additional compute machines using {op-system} 4.4
+boot media.
+
+Adding additional compute machines to your cluster using newer {op-system} boot
+media is not supported.  After adding the compute machines, they will automatically
```
Comment (bobfuru) [inline machine_management/user_infra/adding-vsphere-compute-user-infra.adoc:24]: supported. After
```suggestion
media is not supported. After adding the compute machines, they will automatically
```

**Review by bobfuru — COMMENTED** [review]
Hi @miabbott - a minor nit that I missed before, use a single space between sentences. I'll go ahead and make the change. Otherwise, LGTM.

Let me know if you feel this is ready for me to squash and merge. Thanks!

**Comment by miabbott** [comment]
@bobfuru The content looks good to me; ready for squash and merge

**Comment by bobfuru** [comment]
> @bobfuru The content looks good to me; ready for squash and merge

Well, the button said "Squash and merge" but I guess it kept it as 5 commits ¯\_(ツ)_/¯ but at least it merged. Thanks, @miabbott !

**Pull request merged**